src/coreComponents/physicsSolvers/wavePropagation/pygeosx/mesh.py

```diff
@@ -0,0 +1,297 @@
+'''
+Created on 18/02/2021
+
+@author: macpro
+'''
+import numpy as np
+import pygeosx
+import sys
+from mpi4py import MPI
+import numpy as np
+from functions import *
+    
+
+def initialize_pyMesh(problem):
+    """Get mesh informations from GEOSX mesh for the seismic acquisition
+    
+    Parameters
+    ----------
+    problem :
+        A pygeosx.initialize object
+    
+    Return
+    ------
+    mesh :
+        A Mesh object
+    """
+    
+    node_manager = problem.get_group("domain/MeshBodies/mesh/Level0/nodeManager")
+
+    number_of_node = len(node_manager.get_wrapper("ReferencePosition").value().to_numpy())
+    
+    boundary_node_list = []    
+    for i in range(number_of_node):
+    	if node_manager.get_wrapper("domainBoundaryIndicator").value().to_numpy()[i]==1:
+    	     boundary_node_list.append(i)
+    	     
+
+    elem_manager = problem.get_group("domain/MeshBodies/mesh/Level0/ElementRegions/elementRegionsGroup/Region/elementSubRegions/cb")
+  
+    elem_speed  = problem.get_wrapper("FieldSpecifications/cellVelocity/scale").value()[0]
+    
+    number_of_elem = len(elem_manager.get_wrapper("nodeList").value().to_numpy())
+    
+    nb_node_per_elem = elem_manager.get_wrapper("numNodesPerElement").value()[0]
+    if nb_node_per_elem==8:
+    	disc_ord=1
+    elif nb_node_per_elem==64:
+    	disc_ord=3
+    elif nb_node_per_elem==216:
+    	disc_ord=5
+    
+    
+    Node_list = [Node(node_manager.get_wrapper("ReferencePosition").value().to_numpy()[i], i) for i in range(number_of_node)]
```

Comment:
These in particular looks quite slow. Ideally we should avoid doing any loops in Python that scale with the size of the problem. But if it has to be done, then extract the numpy arrays outside of the loop.

Reply:
Since I haven't defined the seismic acquisition from .sgy files we still need those for now. I modified it to extract the numpy arrays outside the loop.